plugins/twitter/twitter.go

```diff
@@ -40,15 +40,18 @@ func load(consumerKey, consumerSecret, accessToken, accessSecret string, filter
 					switch m := msg.(type) {
 					case *twitter.Tweet:
 						if channel, ok := filter[m.User.IDStr]; ok {
-							if replyChannel, ok := filter[m.InReplyToUserIDStr]; !ok || channel != replyChannel {
-								twitterUrl := fmt.Sprintf("https://twitter.com/%s/status/%s", m.User.ScreenName, m.IDStr)
-								chanId, err := bot.GetChannelId(channel)
-								if err != nil {
-									log.WithField("err", err).Error("unable to find channel.")
+							if m.RetweetedStatus != nil {
+								if replyChannel, ok := filter[m.RetweetedStatus.User.IDStr]; ok || channel == replyChannel {
```

Comment:
add a log in here so that we can get more info if this isn't right

Reply:
also, I think you might want this to be `ok && channel == replyChannel`. If we know about the user that originally tweeted it, we only want to filter it if it was in the same channel.